syntax = "proto3";

package orders;

import "google/protobuf/empty.proto";

message Order {
  string order_id   = 1;
  string user_id    = 2;
  double amount     = 3;
  string description= 4;
  string status     = 5;
}

message CreateOrderRequest {
  string user_id    = 1;
  double amount     = 2;
  string description= 3;
}

message GetOrderRequest {
  string order_id = 1;
}

message ListOrdersResponse {
  repeated Order orders = 1;
}

service OrderService {
  rpc CreateOrder(CreateOrderRequest) returns (Order);
  rpc GetOrder(GetOrderRequest)   returns (Order);
  rpc ListOrders(google.protobuf.Empty) returns (ListOrdersResponse);
}
